### PR TITLE
[Limit orders] Changed error text for UNHANDLED_ERROR

### LINKED ISSUE #2280: Review button blinks with Unhandled error
Reported and related to 
https://cowservices.slack.com/archives/C0375NV72SC/p1681124839607809

From time to time I receive 503 errors in Staging (BARN API) for these requests
 - Request URL: https://barn.api.cow.fi/goerli/api/v1/quot
 - Request URL: https://barn.api.cow.fi/goerli/api/v1/token/0x91056D4A53E1faa1A84306D4deAEc71085394bC8/native_price
and it is resulting in showing 'Unhandled error' message on the UI. In 5-10 seconds requests are updated with 200 code.
Reproducible for all networks.
.

**Expected**: show a loader/other error, but not blink with 'Unhandled error'

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2280

Changed text from 'Unhandled error' to 'Error loading price. Try again later.'

Screenshot 2023-04-11 at 15 29 11

### CODE REVIEW

File: src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export function SwapButton({
 }
 
 const quoteErrorTexts: { [key in GpQuoteErrorCodes]: string } = {
-  [GpQuoteErrorCodes.UNHANDLED_ERROR]: 'Unhandled error',
+  [GpQuoteErrorCodes.UNHANDLED_ERROR]: 'Error loading price. Try again later.',
```
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx:66]: While this fixes the immediate problem, I have questions:

1. Are all unhandled errors price loading errors?
2. Can we handle them somehow as price loading errors don't end up as unhandled errors at this point?

File: src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx [hunk 2]
```diff
@@ -129,11 +129,11 @@ export const limitOrdersTradeButtonsMap: { [key in LimitOrdersFormState]: Button
...
   },
   [LimitOrdersFormState.QuoteError]: ({ quote }: TradeButtonsParams) => {
+    const defaultError = quoteErrorTexts[GpQuoteErrorCodes.UNHANDLED_ERROR]
+
     return (
       <SwapButton disabled={true}>
-        <Trans>
-          {quote.error ? quoteErrorTexts[quote.error.type] || 'Error loading price. Try again later.' : 'Unknown error'}
-        </Trans>
+        <Trans>{quote.error ? quoteErrorTexts[quote.error.type] || defaultError : defaultError}</Trans>
```
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx:136]: What about:

Warning, code not tested!

```suggestion
        <Trans>{(quote.error && quoteErrorTexts[quote.error.type]) || defaultError }</Trans>
```

**Review by alfetopito — APPROVED** [review]

File: src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx (hunk 1 above)
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx:66]: 1. All these errors are from `quote` requests, so, all of them are related to price loading
2. UNHANDLED_ERROR it's any `5xx` HTTP errors

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , great job!

File: src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ export function SwapButton({
 }
 
 const quoteErrorTexts: { [key in GpQuoteErrorCodes]: string } = {
-  [GpQuoteErrorCodes.UNHANDLED_ERROR]: 'Unhandled error',
```
Comment (anxolin) [inline src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx:66]: nitipic: But does this UNHANDLED_ERROR always mean ERROR_LOADING_PRICES? 

If so, maybe we should rename the enum too.

**Pull request merged**